code/game/turfs/simulated/floor/fancy_floor.dm

```diff
@@ -32,20 +32,49 @@
 	icon_state = "grass"
 	floor_tile = /obj/item/stack/tile/grass
 	broken_states = list("sand")
+	var/ore_type = /obj/item/weapon/ore/glass
 
 /turf/open/floor/grass/New()
 	..()
 	spawn(1)
 		update_icon()
 
 /turf/open/floor/grass/attackby(obj/item/C, mob/user, params)
+	if(istype(C, /obj/item/weapon/shovel) && params)
+		new ore_type(src)
+		new ore_type(src) //Make some sand if you shovel grass
+		user << "<span class='notice'>You shovel [src].</span>"
```

Comment:
This becomes "You shovel the snow" and "You shovel the grass patch" and "You shovel the volcanic floor".